src/main/java/org/jabref/model/metadata/MetaData.java

```diff
@@ -245,6 +254,54 @@ public void clearSaveOrderConfig() {
         postChange();
     }
 
+    /**
+     * @param other           the metaData to merge into this one
+     * @param otherFilename   the filename of the other library. Pass "unknown" if not known.
+     * @param allOtherEntries list of all other entries
+     */
+    public void merge(MetaData other, String otherFilename, List<BibEntry> allOtherEntries) {
```

Comment:
Please move these merge metadata methods to the database merger class. There you can also add a merge method operating on `BibDatabaseContext` objects, to have an easy way to merge two databases including all their metadata.